rfcs/005-deal-parameters.md

```diff
@@ -0,0 +1,138 @@
+# RFC-005: Deal Parameters
+
+Author: Aidan Sullivan — @aidan46
+Date: 18/02/25
+
+## Abstract
+
+This RFC proposes features that allow storage providers to set terms for deals so they can automatically accept the deals.
+This will enhance user experience on both the storage provider and storage client side by requiring less intervention on deals that fall between the storage provider's set parameters.
+
+## Introduction
+
+Currently, storage providers accept any deal that adheres to the minimum chain requirements for deals.
+These are just basic requirements that makes sure the deal is valid such as: the start block is before the end block, a deal does not have a start block before the current block and deal durations fall between the chains set deal duration.
+All these requirements make sense but do not give the storage provider any control over what is important for them, profit 💰.
+Allowing the storage provider to set terms for deals they accept will increase transparency on storage deal prices and allows for storage providers to automatically accept deals.
+This, in turn, will improve the storage client experience by making the price of storage more available.
+Storage clients also no longer need to propose a deal and wait for it to be accepted by the storage provider.
+They can propose a deal within the parameters set by the storage provider and be sure that the deal will be accepted.
+
+## Deal Parameters
+
+Storage providers will be able to set their deal parameters by sharing these parameters after registering as a storage provider on the polka-storage chain.
+The deal parameters are not shared during registration because we do not want to force storage providers to use this feature.
+A new extrinsic will be added to the storage provider pallet in the polka-storage chain to allow storage providers to advertise their deal parameters on-chain.
+Storage clients will be able to query deal parameters of all storage providers so they can find the best deal for them.
+Deal parameters could have endless options, a good starting point is for storage providers to set a lower and upper bound for storage price per block and the duration of a deal.
+These parameters can be expanded to include things such as collateral bounds and different price bounds depending on the size of the deal.
```

Comment:
You should note that the provider should be able to register multiple parameters at the same time.
Example: for deals that have a really long duration, they may want to do a lower price